src/devices/Mcp3008/Mcp3008.cs

```diff
@@ -56,88 +60,102 @@ public void Dispose()
             }
         }
 
-        public int Read(int adc_channel)
+        public int Read(int channel, InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
         {
-            if (adc_channel < 0 || adc_channel > 7)
+            if (channel < 0 || channel > 7)
             {
                 throw new ArgumentException("ADC channel must be within 0-7 range.");
             }
 
             if (_protocol == CommunicationProtocol.Spi)
             {
-                return ReadSpi(adc_channel);
+                return ReadSpi(channel, inputConfiguration);
             }
             else
             {
-                return ReadGpio(adc_channel);
+                return ReadGpio(channel, inputConfiguration);
             }
         }
 
+        private static byte GetConfigurationBits(int channel,InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
+        {
+            int configurationBits;
+
+            if (inputConfiguration == InputConfiguration.SingleEnded)
+            {
+                configurationBits = 0b00011000 | channel;
+            }
+            else
+            {
+                configurationBits = 0b00010000 | channel;
+            }
+
+            configurationBits = configurationBits << 3;
+            return (byte)configurationBits;
```

Comment:
would it be clearer something like
```c#
        private static byte GetConfigurationBits(int channel, InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
             {

            byte configurationBits = 0b10000000;

            if (inputConfiguration == InputConfiguration.SingleEnded)
            {
                configurationBits &= 0b11000000;
            }

            return configurationBits;
        }
```
?

Reply:
@danmosemsft The ConfigurationBits are made of:

`<Start bit><Mode bit><3 channel bits><Remaining 3 unused bits>`

So we need to include the channel value before the bit shift.  Your example provided an idea where the SingleEnded mode is most likely going to be the popular Read choice.  So how about this...

```
private static byte GetConfigurationBits(int channel, InputConfiguration inputConfiguration)
{
    int configurationBits = (0b00011000 | channel) << 3;

    if (inputConfiguration == InputConfiguration.Differential)
    {
        configurationBits &= 0xBF;  // Clear mode bit.
    }
            
    return (byte)configurationBits;
}
```
**NOTE**: I removed the optional argument value as this method is private and should be only called by ReadGpio() and ReadSpi().  Also, we don't check correct channel range here as it's expected in main Read() method call.

Reply:
My bad, I somehow dropped the channel part. What you wrote looks nice to me although I think binary rather than hex format is clearer here. I defer to @joperezr as I'm jsut a visitor here 😀

Reply:
I like binary format, as well.  IMHO, it is better to read the bit-level, but it could sometimes get a little long based on number of bytes.

Reply:
I do agree now with both of you. Let's use binary and let's try to use '_' for every 4 bits for readibility.

Reply:
@joperezr Are you suggesting to update all **0x** values to **0b** format in the binding and sample projects?  I can, just wanted to make sure you weren't just talking about this particular area.